inst/include/dust/random/cuda_compatibility.hpp

```diff
@@ -39,4 +41,23 @@
 #define SYNCWARP
 #endif
 
+namespace dust {
+namespace utils {
+
+// We cannot throw errors in GPU code, we can only send a trap signal,
+// which is unrecoverable.
+__host__ __device__
+inline void fatal_error(const char *message) {
```

Comment:
this might benefit from `__forceinline__`, but maybe the compiler works that out anyway

Reply:
Actually thinking about how it's called, probably doesn't matter

Reply:
The `inline` there is only really necessary because otherwise it causes problems at linkage :)